app/workers/retirement_worker.rb

```diff
@@ -0,0 +1,16 @@
+class RetirementWorker
+  include Sidekiq::Worker
+
+  sidekiq_options queue: :high, retry: 3
+  
+  def perform(subject_id, workflow_id)
+    sms = SetMemberSubject.by_subject_workflow(subject_id, workflow_id)
+    if sms.retire?
+      sms.retired!
+      Cellect::Client.connection
+        .remove_subject(subject_id,
+                        workflow_id: workflow_id,
+                        group_id: sms.subject_set_id)
```

Comment:
Don't we want to retire all sms's for this workflow in the case where the subject exists across subject_sets for the same workflow?